pandas/core/dtypes/cast.py

```diff
@@ -1026,3 +1027,52 @@ def find_common_type(types):
             return np.object
 
     return np.find_common_type(types, [])
+
+
+def maybe_cast_to_integer(arr, dtype):
```

Comment:
@jreback : One wonders if we could just generalize to accept scalar as well as `array` (as consistent with other casting functions in this module) and keep the name as is.